osu.Game/Rulesets/Objects/Drawables/DrawableHitObject.cs

```diff
@@ -394,6 +390,15 @@ private void onDefaultsApplied(HitObject hitObject)
             Debug.Assert(Entry != null);
             Apply(Entry);
 
+            // Applied defaults indicate a change in hit object state.
+            // We need to update the judgement result time to the new end time
+            // and update state to ensure the hit object fades out at the correct time.
+            if (Result is not null)
```

Comment:
Given that `Result` is proxied to `Entry?.Result`, and `Entry` is asserted non-null, this seems like a branch that will always be taken, so may as well get rid of the conditional.

Reply:
The `Result` itself inside the `HitObjectLifetimeEntry` is nullable.